src/main/java/com/google/gwt/site/webapp/client/GWTProjectEntryPoint.java

```diff
@@ -122,68 +144,96 @@ public void f(Element e) {
   private void openMenu() {
     // close all submenus
     // todo hide first anchor with css
-    $("#submenu > nav > ul > li").hide().children("a").hide();
+    forEach("#submenu > nav > ul > li", el -> {
+      el.style.display = "none";
+      forEachChild(el, "a", child -> child.style.display = "none");
+    });
 
     String path = Window.Location.getPath();
-    GQuery selectedItem = $("#submenu a[href='" + path + "']")
-        .filter(new Predicate() {
-          @Override
-          public boolean f(Element e, int index) {
-            return !Style.Display.NONE.getCssName().equals(e.getStyle().getDisplay());
-          }
-        })
-        .eq(0);
+    HTMLElement[] container = new HTMLElement[1];
+    forEach("#submenu a[href='" + path + "']", el -> {
+      if (!el.style.display.equals("none")) {
+        container[0] = el;
+      }
+    });
+    HTMLElement selectedItem = container[0];
+    String mainNavigationHref;
+    String mainTitle;
+    if (selectedItem != null) {
 
-    showBranch(selectedItem);
+      showBranch(selectedItem);
 
-    GQuery liParents = selectedItem
-        .parentsUntil("#submenu")
-        .filter("li");
-    GQuery subMenuItem = liParents.last();
+      List<HTMLElement> liParents = parentMenuItems(selectedItem);
 
-    subMenuItem.show();
-    String mainNavigationHref = subMenuItem.children("a").attr("href");
+      HTMLElement subMenuItem = liParents.get(liParents.size() - 1);
+
+      subMenuItem.style.display = "";
+      forEach("#submenu .active", el -> {
+        if (!liParents.contains(el)) {
+          el.classList.remove("active");
+        }
+      });
+      liParents.add(selectedItem);
+      liParents.forEach(el -> {
+        if (el != selectedItem.parentElement) {
+          el.classList.add("active");
+        }
+      });
+      mainNavigationHref = subMenuItem.querySelector("a").getAttribute("href");
+      mainTitle = subMenuItem.querySelector("a").textContent + " - " + selectedItem.textContent;
+    } else {
+      mainNavigationHref = Window.Location.getPath();
+      mainTitle = isOverviewPage(mainNavigationHref) ? "Overview" : "Project";
```

Comment:
This looks like a change in behavior? I'm not necessarily opposed, but it does make reviewing this giant rewrite a little trickier.

Reply:
Yes, since the `GQuery` selectors are NPE-safe I had to add the `if (selectedItem != null)` check and then it felt weird to replicate the buggy behavior in the `else` branch :shrug:

This should be the only behavior change though.